engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
I would prefer to have a separate bug and PR for changing what window Id is safe to purge.